internal/wireguard/run.go

```diff
@@ -123,7 +123,14 @@ func (w *Wireguard) Run(ctx context.Context, waitError chan<- error, ready chan<
 		waitError <- fmt.Errorf("%w: %s", ErrRuleAdd, err)
 		return
 	}
+	ruleCleanup6, err := w.addRule6(
+		w.settings.RulePriority, w.settings.FirewallMark)
+	if err != nil {
+		waitError <- fmt.Errorf("%w: %s", ErrRuleAdd, err)
+		return
+	}
 	closers.add("removing rule", stepOne, ruleCleanup)
+	closers.add("removing rule", stepOne, ruleCleanup6)
```

Comment:
Move this ipv6 stuff in the `doIPv6` if block instead? It's fine if it executes before the ipv4 add rule AFAIK